geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,201 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
-            else:
-                plugin_entry = [
-                    interface_registry[plugin_key]["package"],
-                    interface.name,
-                    interface.interface_type,
-                    interface_registry[plugin_key]["family"],
-                    plugin_key,
-                    "N/A",
-                    interface_registry[plugin_key]["relpath"],
-                ]
-                table_data.append(plugin_entry)
-        headers = [
-            "GeoIPS Package",
-            "Interface",
-            "Interface Type",
-            "Family",
-            "Plugin Name",
-            "Source Names",
-            "Relative Path",
-        ]
+        default_headers = {
+            "package": "GeoIPS Package",
+            "interface": "Interface Name",
+            "plugin_type": "Interface Type",
+            "family": "Family",
+            "plugin_name": "Plugin Name",
+            "source_names": "Source Names",
+            "relpath": "Relative Path",
+        }
+        headers = self._get_headers_by_command(args, default_headers)
+        table_data = self._generate_table_data_by_interface(
+            interface,
+            interface_registry,
+            headers,
+        )
         print(
             tabulate(
                 table_data,
-                headers=headers,
+                headers=headers.values(),
                 tablefmt="rounded_grid",
                 maxcolwidths=self.terminal_width // len(headers),
             )
         )
+
+    def _generate_table_data_by_interface(self, interface, interface_registry, headers):
+        """Generate the table data needed for output in '_print_plugins_short_format.
+
+        Given a certain interface and a list of headers, generate a list of table data
+        used for outputting information used by 'geoips list interface' and
+        'geoips list plugins'.
+
+        Parameters
+        ----------
+        interface: geoips.interfaces.<interface_type>
+            - The interface object we will be gathering data from.
+        interface_registry: dict
+            - The plugin registry associated with the given interface.
+        headers: dict
+            - Dictionary of strings representing the key-value mapping of the
+              headers we'd like to output.
+
+        Returns
+        -------
+        table_data: 2D List of Strings
+            - The corresponding data we will output, where each sub-list is ordered
+              by headers.
+        """
+        table_data = []
+        if interface.name == "products":
+            for plugin_key in sorted(interface_registry.keys()):
+                product_dict = interface_registry[plugin_key]
+                table_data += self._get_entry(product_dict, headers)
+        else:
+            table_data += self._get_entry(interface_registry, headers)
+        return table_data
+
+    def _get_entry(self, plugin_dict, headers):
```

Comment:
Is this the correct name for this function? It looks like it returns the entire table rather than a single row so does "entry" make sense here?

Reply:
I think we should leave it as ``get_entry`` or change it to ``get_plugin_entry``. This would only be called ``get_plugin_entries`` for product plugins currently, but would change to ``get_entry`` / ``get_plugin_entry`` once the changes in #474 come into effect.

Reply:
Since they would not be ``abi --> product_plugins``, and would be changed to `single `abi.Infrared``entries and so on.